src/Tribe/Integrations/WPML/Linked_Posts.php

```diff
@@ -96,39 +127,41 @@ public function filter_tribe_events_linked_posts_query( $results = null, array $
 		// IDs only and drop the order to avoid wasting time on something we'll account for later
 		$sub_query_args = array_merge( $args, array( 'fields' => 'ids', 'order' => false ) );
 
-		$linked_posts_ids = $this->get_curent_language_linked_posts_ids( $sub_query_args );
+		$linked_posts_ids = $this->get_current_language_linked_posts_ids( $sub_query_args );
 
 		$default_lang_linked_posts_ids = $this->get_default_language_linked_post_ids( $sub_query_args );
 
 		$linked_posts_ids = array_merge( $default_lang_linked_posts_ids, $linked_posts_ids );
 
-		// run this query to keep the specified `orderby`
-		$linked_posts = get_posts( array_merge( $args, array( 'post__in' => $linked_posts_ids ) ) );
+		if ( false !== $post__not_in ) {
+			$linked_posts_ids = array_diff( $linked_posts_ids, $post__not_in );
+		}
 
-		$sitepress->switch_lang( ICL_LANGUAGE_CODE );
+		if ( empty( $linked_posts_ids ) ) {
+			return $linked_posts = array();
+		} else {
+			// run this query to keep the specified `orderby`
+			$linked_posts = get_posts( array_merge( $args, array( 'post__in' => $linked_posts_ids ) ) );
+		}
+
+		$this->cache[ $cache_key ] = $linked_posts;
 
 		return $linked_posts;
 	}
 
 	/**
-	 * @param int $id The post ID
+	 * Returns a list of post IDs of linked posts for the current language.
 	 *
-	 * @return bool `true` if the post lacks a WPML managed translation, `false` if the post has a WPML managed translation.
-	 */
-	protected function is_not_translated( $id ) {
-		/** @var SitePress $sitepress */
-		global $sitepress;
-		$translation_id = $sitepress->get_object_id( $id, 'post', true, ICL_LANGUAGE_CODE );
-
-		return empty( $translation_id ) || $translation_id == $id;
-	}
-
-	/**
 	 * @param array $args An array WP_Query arguments
 	 *
 	 * @return array An array of linked posts filtered by the current language
 	 */
-	protected function get_curent_language_linked_posts_ids( array $args ) {
+	protected function get_current_language_linked_posts_ids( array $args ) {
+		$cache_key = $this->cache->make_key( func_get_args(), 'current_language_linked_post_ids' );
```

Comment:
You cannot use `func_get_args()` directly into a function/method call, you need to save it in a variable first. PHP 5.2 compat.